### PR TITLE
[Shop] Change payment method on an order

### PR DESCRIPTION
**Pull request by tuka217**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Related tickets |  #5415
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Ui/CheckoutContext.php [hunk 1]
```diff
@@ -910,6 +910,22 @@ public function iCompleteAddressingStepWithEmail($email, AddressInterface $addre
     }
 
     /**
+     * @When I choose payment method :paymentMethodName
+     */
+    public function iChoosePaymentMethod1($paymentMethodName)
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Ui/CheckoutContext.php]: ``iChoosePaymentMethod``

File: src/Sylius/Behat/Page/Shop/Checkout/ThankYouPage.php [hunk 2]
```diff
@@ -56,6 +57,20 @@ public function waitForResponse($timeout, array $parameters = [])
     }
 
     /**
+     * {@inheritdoc}
+     */
+    public function choosePaymentMethod($paymentMethodName)
+     {
```
Comment (Zales0123) [inline src/Sylius/Behat/Page/Shop/Checkout/ThankYouPage.php]: cs

File: src/Sylius/Bundle/ShopBundle/Resources/config/routing/order.yml [hunk 3]
```diff
@@ -20,3 +20,18 @@ sylius_shop_order_after_pay:
         _sylius:
             redirect:
                 route: sylius_shop_checkout_thank_you
+
+sylius_shop_order_payment_method_update:
+    path: /{orderId}/payment_method/edit
```
Comment (Zales0123) [inline src/Sylius/Bundle/ShopBundle/Resources/config/routing/order.yml]: ``payment_method`` -> ``payment-method``

**Comment by Zales0123** [comment]
@tuka217 rebase required ;) And there is a typo in PR title ``a order`` -> ``an order`` 😄  🐃

File: features/checkout/paying_for_order/changing_method_after_checkout.feature [hunk 4]
```diff
@@ -9,15 +9,18 @@ Feature: Changing the method on the order payment page
...
-    @todo
+    @ui
     Scenario: Retrying the payment with offline payment
         Given I added product "PHP T-Shirt" to the cart
-        And I confirmed my order selecting PayPal Checkout Express payment method
-        But I failed to pay
-        And I am redirected back to the order payment page
-        When I try to pay again with offline method
-        Then I should be redirected back to the thank you page
+        And I proceed selecting "PayPal Express Checkout" payment method
```
Comment (lchrusciel) [inline features/checkout/paying_for_order/changing_method_after_checkout.feature]: `proceed` -> `proceeded`

File: src/Sylius/Behat/Context/Ui/CheckoutContext.php [hunk 5]
```diff
@@ -921,6 +910,22 @@ public function iCompleteAddressingStepWithEmail($email, AddressInterface $addre
...
+    {
+        $this->thankYouPage->choosePaymentMethod($paymentMethodName);
+    }
+
+    /**
+     * @Given I confirm my changes
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Ui/CheckoutContext.php]: `@Given` -> `@When`

File: src/Sylius/Behat/Service/Mocker/MockerInterface.php [hunk 6]
```diff
@@ -32,4 +32,6 @@ public function mockCollaborator($className);
      * @return MockInterface
      */
     public function mockService($serviceId, $className);
+
+    public function unmockAllServices();
```
Comment (Zales0123) [inline src/Sylius/Behat/Service/Mocker/MockerInterface.php]: Is it needed? I think it has been removed eventually.

File: src/Sylius/Behat/Service/Mocker/PaypalApiMockerInterface.php [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Service\Mocker;
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+interface PaypalApiMockerInterface
```
Comment (Zales0123) [inline src/Sylius/Behat/Service/Mocker/PaypalApiMockerInterface.php]: In fact whole ``PaypalApiMockerInterface`` and related code.

**Comment by Zales0123** [comment]
@tuka217 Last fixes + rebase and I think it'll be ready to go ;)

File: src/Sylius/Behat/Context/Ui/CheckoutContext.php [hunk 8]
```diff
@@ -881,6 +889,14 @@ private function createDefaultAddress()
     }
 
     /**
+     * @Given I confirm my changes
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Ui/CheckoutContext.php]: `Given` -> `When`

**Comment by pjedrzejewski** [comment]
Thank you Ania, nice work!

**Pull request merged**